gpt_index/readers/weaviate/data_structs.py

```diff
@@ -224,7 +228,12 @@ def _from_gpt_index(cls, client: Any, node: Node, class_prefix: str) -> str:
         # TODO: account for existing nodes that are stored
         node_id = get_new_id(set())
         class_name = cls._class_name(class_prefix)
-        client.batch.add_data_object(node_dict, class_name, node_id, vector)
+
+        # if batch object is provided (via a contexxt manager), use that instead
+        if batch is not None:
+            batch.add_data_object(node_dict, class_name, node_id, vector)
+        else:
+            client.batch.add_data_object(node_dict, class_name, node_id, vector)
```

Comment:
@jerryjliu Can we assume that this function (`_from_gpt_index`) will always be called under the batch context manager? If not, then, the the default batch size is None which means manual batching will be used. The implication is that the user will need to explicitly call `batch.flush()` themselves.

Reply:
that's a fair point. atm yes, this will always be called under the context manager. there is also a `from_gpt_index` method which calls `_from_gpt_index` and will flush the batch; atm we don't use it but we might later

Reply:
alright, then this looks good!